utils/shared.js

```diff
@@ -797,6 +797,80 @@ const redactEmailLoginInfo = (participantEmail) => {
 
 const redactPhoneLoginInfo = (participantPhone) => "***-***-" + participantPhone.slice(-4);
 
+// Note: '223999569' is Biohazard bag (mouthwash) scan, '787237543' is Biohazard bag (blood/urine) scan, '522094118' is Orphan bag scan. These are not tubes.
+const tubeConceptIds = [
+    '143615646', // Mouthwash tube 1
+    '232343615', // Serum separator tube 4
+    '299553921', // Serum separator tube 1
+    '376960806', // Serum separator tube 3
+    '454453939', // EDTA tube 1
+    '589588440', // Serum separator tube 5
+    '652357376', // ACD tube 1
+    '677469051', // EDTA tube 2
+    '683613884', // EDTA tube 3
+    '703954371', // Serum separator tube 2
+    '838567176', // Heparin tube 1
+    '958646668', // Heparin tube 2
+    '973670172', // Urine tube 1
+];
+
+// Extract all the collectionIds from a list of boxes, return an array of unique collectionIds.
+// Bag types: 787237543 (Biohazard Blood/Urine), 223999569 (Biohazard Mouthwash), 522094118 (Orphan)
+const extractCollectionIdsFromBoxes = (boxesList) => {
+    const { bagConceptIDs } = require('./shared');
+    const collectionIdSet = new Set();
+    for (const box of boxesList) {
+        for (const bag of bagConceptIDs) {
+            if (box[bag]) {
+                const bagId = box[bag]['787237543'] || box[bag]['223999569'] || box[bag]['522094118'];
+                const collectionId = bagId.split(' ')[0];
+                if (collectionId) {
+                    collectionIdSet.add(collectionId);
+                }
+            }
+        }
+    }
+    return Array.from(collectionIdSet);
+}
+
+// process fetched specimen collections, filter out tubes that are not received on the receivedTimestamp day.
+// return specimen data array with modified tube data
+const processSpecimenCollections = (specimenCollections, receivedTimestamp) => {
+    let tubeCount = 0;
+    const specimenDataArray = [];
+
+    for (const specimenCollection of specimenCollections) {
+        let hasSpecimens = false;
+
+        const filteredSpecimens = tubeConceptIds.reduce((acc, key) => {
+            const tube = specimenCollection[key];
+            if (tube && tube['926457119'] === receivedTimestamp) {
+                tubeCount++;
```

Comment:
I'm not seeing `tubeCount` being use.

Reply:
Updated. Good catch - that was from testing. Removed.